x-pack/plugins/apm/public/components/app/service_inventory/index.tsx

```diff
@@ -12,21 +12,20 @@ import uuid from 'uuid';
 import { useAnomalyDetectionJobsContext } from '../../../context/anomaly_detection_jobs/use_anomaly_detection_jobs_context';
 import { useLegacyUrlParams } from '../../../context/url_params_context/use_url_params';
 import { useLocalStorage } from '../../../hooks/use_local_storage';
-import { useAnyOfApmParams } from '../../../hooks/use_apm_params';
+import { useApmParams } from '../../../hooks/use_apm_params';
 import { FETCH_STATUS, useFetcher } from '../../../hooks/use_fetcher';
 import { useTimeRange } from '../../../hooks/use_time_range';
 import { SearchBar } from '../../shared/search_bar';
 import { getTimeRangeComparison } from '../../shared/time_comparison/get_time_range_comparison';
 import { ServiceList } from './service_list';
 import { MLCallout, shouldDisplayMlCallout } from '../../shared/ml_callout';
+import { joinByKey } from '../../../../common/utils/join_by_key';
 
 const initialData = {
   requestId: '',
-  mainStatisticsData: {
-    items: [],
-    hasHistoricalData: true,
-    hasLegacyData: false,
-  },
+  items: [],
+  hasHistoricalData: true,
+  hasLegacyData: false,
```

Comment:
I think `hasLegacyData` is leftover and should have been removed in https://github.com/elastic/kibana/pull/125962.
`hasHistoricalData` I'm not sure about - I don't see any references to it anywhere. Was that removed accidentally? Or perhaps we replaced that with something else when we introduced the new Observability "no data" screen?

tldr: I think both of these should be removed but please double check

Reply:
`hasHistoricalData` was removed in https://github.com/elastic/kibana/pull/111630. It is no longer needed.

Reply:
However, it looks like the message is now simply saying "No services found":

https://github.com/elastic/kibana/blob/8b82657d46e18920e3a1acc3133f10f78e25a6f4/x-pack/plugins/apm/public/components/app/service_inventory/index.tsx#L127-L138

It should instead say something like "No data found for selected period. Please adjust your time range and other 
possible filters".

(fyi @MiriamAparicio)